Flaky Test `DeploymentClusteredTest.shouldRedistributeDeploymentWhenDeploymentPartitionIsRestarted`

**Summary**

- How often does the test fail? [once](https://github.com/camunda/zeebe/actions/runs/8552447128/attempts/1#summary-23433567011)
- Does it block your work? no
- Do we suspect that it is a real failure? IDK

**Failures**

<details><summary>Example assertion failure</summary>
<pre>
shouldRedistributeDeploymentWhenDeploymentPartitionIsRestarted (Flaky Test): io.camunda.zeebe.it.clustering.DeploymentClusteredTest
java.lang.AssertionError: 

Expecting Optional to contain a value but it was empty.
	at io.camunda.zeebe.it.clustering.DeploymentClusteredTest.shouldRedistributeDeploymentWhenDeploymentPartitionIsRestarted(DeploymentClusteredTest.java:132)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.lang.Thread.run(Thread.java:1583)
</pre>
</details>

**Hypotheses**
I don't have an idea

**Logs**

<details><summary>Logs</summary>

[LOGS](https://drive.google.com/file/d/1ktAqb-OAA14cOd6Lo9z3sNFkf_cSfjvC/view?usp=drive_link)
</details>

--- a/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/DeploymentClusteredTest.java
+++ b/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/DeploymentClusteredTest.java
@@ -123,13 +123,7 @@ public final class DeploymentClusteredTest {
             .addProcessModel(PROCESS, "process.bpmn")
             .send()
             .join();
-    final var processDefinitionKey = deploymentEvent.getKey();
-
-    assertThat(
-            RecordingExporter.commandDistributionRecords(CommandDistributionIntent.ACKNOWLEDGED)
-                .withDistributionPartitionId(3)
-                .findFirst())
-        .isPresent();
+    final var deploymentKey = deploymentEvent.getKey();
 
     clusteringRule.stopBroker(deploymentPartitionLeader);
     adminServiceLeaderTwo.resumeStreamProcessing();
@@ -139,10 +133,11 @@ public final class DeploymentClusteredTest {
     clusteringRule.getClock().addTime(DEPLOYMENT_REDISTRIBUTION_INTERVAL);
 
     // then
-    clientRule.waitUntilDeploymentIsDone(processDefinitionKey);
+    clientRule.waitUntilDeploymentIsDone(deploymentKey);
 
     assertThat(
             RecordingExporter.commandDistributionRecords(CommandDistributionIntent.ACKNOWLEDGED)
+                .withRecordKey(deploymentKey)
                 .withDistributionPartitionId(2)
                 .findFirst())
         .isPresent();

